x-pack/plugins/apm/public/components/app/backend_inventory/backend_inventory_dependencies_table/index.tsx

```diff
@@ -10,33 +10,30 @@ import { i18n } from '@kbn/i18n';
 import React from 'react';
 import { useUiTracker } from '../../../../../../observability/public';
 import { getNodeName, NodeType } from '../../../../../common/connections';
-import { useLegacyUrlParams } from '../../../../context/url_params_context/use_url_params';
 import { useApmParams } from '../../../../hooks/use_apm_params';
+import { useComparison } from '../../../../hooks/use_comparison';
 import { useFetcher } from '../../../../hooks/use_fetcher';
 import { useTimeRange } from '../../../../hooks/use_time_range';
 import { BackendLink } from '../../../shared/backend_link';
 import { DependenciesTable } from '../../../shared/dependencies_table';
-import { getTimeRangeComparison } from '../../../shared/time_comparison/get_time_range_comparison';
 
 export function BackendInventoryDependenciesTable() {
   const {
-    urlParams: { comparisonEnabled, comparisonType },
-  } = useLegacyUrlParams();
-
-  const {
-    query: { rangeFrom, rangeTo, environment, kuery },
+    query: {
+      rangeFrom,
+      rangeTo,
+      environment,
+      kuery,
+      comparisonEnabled,
+      comparisonType,
+    },
   } = useApmParams('/backends');
 
   const { start, end } = useTimeRange({ rangeFrom, rangeTo });
 
   const trackEvent = useUiTracker();
 
-  const { offset } = getTimeRangeComparison({
-    start,
-    end,
-    comparisonEnabled,
-    comparisonType,
-  });
+  const { offset } = useComparison();
```

Comment:
I strongly recommend using `getTimeRangeComparison` here still. `useComparison` (without arguments) breaks any type safety that we have from the route params.

Reply:
Can you please explain further? It seems that `rangeFrom`, `rangeTo`,  `comparisonEnabled` and `comparisonType` are retrieved in a similar way from `useComparison`

Reply:
Giorgos and I had a call to talk about it already, he would look into using `getTimeRangeComparison` where possible. IMHO we should avoid depending on runtime checks, because it means more (automated or manual) tests will be necessary - it won't be the first time we've forgot about requiring certain parameters on a route. rangeFrom/rangeTo are pretty widely available, but I think these principles should be applied consistently where possible.

Reply:
As discussed with @dgieselaar, `useComparison` hook was removed in https://github.com/elastic/kibana/pull/126567/commits/367fa6b82f159cdf8b0cb191cd074dbd129d5c0d. I've suggested to pass the route in the hook but we would end up handling it differently than `useTimeRange`.